File: 백준/Silver/15654. N과 M （5）/N과 M （5）.swift
var nums = readLine()!.split(separator: " ").map {Int(String($0))!}
var numbers = readLine()!.split(separator: " ").map {Int(String($0))!}
numbers.sort()
var selected = [Int]()
var answer = ""

func DFS() {
    if selected.count == nums[1] {
        answer += selected.map { String($0) }.joined(separator: " ") + "\n"
        return
    }
    for num in 0..<nums[0] {
        if !selected.contains(numbers[num]){
            selected.append(numbers[num])
            DFS()
            selected.removeLast()
        }
    }
}

DFS()
print(answer)

File: 백준/Silver/15654. N과 M （5）/N과 M （5）.py
n, m = map(int, input().split())
nums = list(map(int, input().split()))
nums.sort()

lst = []

def DFS():
    if len(lst) == m:
        print(" ".join(lst))
        return
    for i in nums:
        if str(i) not in lst:
            lst.append(str(i))
            DFS()
            lst.pop()

DFS()